backend/classification/features/extraction.py

```diff
@@ -0,0 +1,76 @@
+"""Feature extraction tools based off a two channel EEG recording"""
+import numpy as np
+
+from classification.config.constants import (
+    EEG_CHANNELS,
+    AGE_FEATURE_BINS,
+    DATASET_SAMPLE_RATE,
+)
+from classification.features.constants import DATASET_HIGH_PASS_FREQ
+from classification.features.pipeline import get_feature_union
+from classification.features.utils import (
+    drop_other_channels,
+    crop_raw_data,
+    convert_to_epochs,
+)
+
+
+def get_eeg_features(raw_data, in_bed_seconds, out_of_bed_seconds):
+    """Returns the continuous feature matrix
+    Input
+    -------
+    raw_signal: MNE.Raw object with signals with or without annotations
+    in_bed_seconds: timespan, in seconds, from which the subject started
+        the recording and went to bed
+    out_of_bed_seconds: timespan, in seconds, from which the subject
+        started the recording and got out of bed
+
+    Returns
+    -------
+    Array of size (nb_epochs, nb_continuous_features)
+    """
+    features_file = []
+
+    for channel in EEG_CHANNELS:
+        chan_data = drop_other_channels(raw_data.copy(), channel)
+        chan_data = crop_raw_data(chan_data, in_bed_seconds, out_of_bed_seconds)
+        chan_data = chan_data.resample(DATASET_SAMPLE_RATE)
```

Comment:
En relisant la documentation, il semble que la deuxième option est plus appropriée dans notre cas

1. resample the data after epoching, and make your epochs long enough that **edge effects from the filtering do not affect the temporal span of the epoch that you hope to analyze / interpret;** or
2. perform resampling on the Raw object and its corresponding Events array simultaneously so that they stay more or less in synch. This can be done by passing the Events array as the events parameter to raw.resample().
[source](https://mne.tools/dev/auto_tutorials/preprocessing/plot_30_filtering_resampling.html)

Si l'on downsample chaque epoch, on introduit des edges effects dans chacune d'entre elles. Comme on n'a pas de baseline avant et après qu'on peut juste enlever, et donc que c'est l'epoch au complet qui nous intéresse, je ne crois pas qu'on peut prendre cette approche.
Il faudrait donc resample l'objet raw, et en lui passant nos annotations, qu'il ajusterait...

Autre chose que l'on pourrait considérer, serait de ne tout simplement pas resample. On fait l'extraction de features sur nos epochs originales. C'est sur que ça serait plus lent et reste à voir si ça donne des bons résultats...